src/autogpt_plugins/email/email_plugin/email_plugin.py

```diff
@@ -311,3 +312,49 @@ def clean_email_body(email_body):
     email_body = re.sub(r"http\S+", "", email_body)
 
     return email_body
+
+def read_summarized_emails(
+        imap_folder: str = "inbox", imap_search_command: str = "UNSEEN", limit: int = 5,
+        page: int = 1):
+    """Read emails from an IMAP mailbox and return summarized results
+
+    This function reads emails from a specified IMAP folder, using a given IMAP search command, limits, and page numbers.
+    It returns a list of emails with their details, including the sender, recipient, date, CC, subject, and summarized 
+    message body.
+
+    Args:
+        imap_folder (str, optional): The name of the IMAP folder to read emails from. Defaults to "inbox".
+        imap_search_command (str, optional): The IMAP search command to filter emails. Defaults to "UNSEEN".
+        limit (int, optional): Number of email's the function should return. Defaults to 5 emails.
+        page (int, optional): The index of the page result the function should resturn. Defaults to 0, the first page.
+
+    Returns:
+        str: A list of dictionaries containing email details if there are any matching emails. Otherwise, returns
+             a string indicating that no matching emails were found."""
+
+    # Retrieve emails
+    emails = read_emails(imap_folder, imap_search_command, limit, page)
+
+    # Create an OpenAI connection using the Auto-GPT session's API key
+    openai.api_key = os.getenv("OPENAI_API_KEY")
```

Comment:
I can do this. I am not as familiar with the whole AutoGPT architecture; if you know off the top of your head how to access the OpenAI session from a plugin, it would save me some time. If not, I can dig through the code.

Reply:
Some changes to configs are coming that should make that more accessible

Reply:
Ok, I'll hold off in implementing this until the changes are part of master.